lib/Common/Codex/Utf8Helper.h

```diff
@@ -141,12 +132,43 @@ namespace utf8
             utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(remDestString, remSourceString, (LPCUTF8) sourceString + cbSourceString, DecodeOptions::doAllowInvalidWCHARs);
             Assert(destString[cchDestString] == 0);
             static_assert(sizeof(utf8char_t) == sizeof(char), "Needs to be valid for cast");
-            *destStringPtr = destString;
             *destCount = cchDestString;
         }
+
         return S_OK;
     }
 
+    ///
+    /// Use the codex library to encode a UTF8 string to UTF16.
+    /// The caller is responsible for freeing the memory, which is allocated
+    /// using Allocator.
+    /// The returned string is null terminated.
+    ///
+    template <typename AllocatorFunction>
+    HRESULT NarrowStringToWide(_In_ AllocatorFunction allocator,_In_ LPCSTR sourceString,
+        size_t sourceCount, _Out_ LPWSTR* destStringPtr, _Out_ size_t* destCount, size_t* allocateCount = nullptr)
+    {
+        size_t cbDestString = (sourceCount + 1) * sizeof(WCHAR);
+        if (cbDestString < sourceCount) // overflow ?
+        {
+            return E_OUTOFMEMORY;
+        }
+
+        WCHAR* destString = (WCHAR*)allocator(cbDestString);
+        if (destString == nullptr)
+        {
+            return E_OUTOFMEMORY;
+        }
+
+        if (allocateCount != nullptr) *allocateCount = cbDestString;
+
+        *destStringPtr = destString;
+        HRESULT result = NarrowStringToWideNoAlloc(sourceString, sourceCount, destString, sourceCount, destCount);
```

Comment:
I believe that the `destBufferCount` parameter here should be `sourceCount+1` to allow for the null